### PR TITLE
Piece::Rook

### PR DESCRIPTION
**Pull request by mjhoy**

### CODE REVIEW

File: src/state.rs [hunk 1]
```diff
@@ -138,6 +278,26 @@ mod test {
         assert!(black_move.can_move_pseudo(b7, b6));
     }
 
+    #[test]
+    fn test_rook_moves() {
+        let board = test_simple_board_for_piece_lateral_king(Piece::Rook);
+        let a4 = Pos { rank: 3, file: 0 };
```
Comment (paulkernfeld) [inline src/state.rs]: If you wanted to get rid of these, an MVP would be to just make a constant for every position on a normal chess board.

File: src/state.rs [hunk 2]
```diff
@@ -108,20 +165,103 @@ impl State {
...
 
     fn test_board() -> Board {
         Board::initial()
     }
 
+    fn test_simple_board_for_piece_lateral_king(piece: Piece) -> Board {
+        // White king at d4, black king at b7, and white's variable
+        // piece at c4. E.g., the queen on this board:
+        // https://lichess.org/analysis/standard/8/1k6/8/8/2QK4/8/8/8_w_-_-
+        let inner = vec![
```
Comment (paulkernfeld) [inline src/state.rs:179]: Feature request: parsing a board from a Unicode raw text string.

**Review by paulkernfeld — COMMENTED** [review]
👍

File: src/state.rs (hunk 1 above)
Comment (mjhoy) [reply src/state.rs]: good point! bbd2df7786730ea00e99eef07d962986acde840f

File: src/state.rs (hunk 2 above)
Comment (mjhoy) [reply src/state.rs:179]: eventually i would like to parse https://en.wikipedia.org/wiki/Forsyth–Edwards_Notation

**Pull request merged**